File: 4135-concatenate-non-zero-digits-and-multiply-by-sum-i/4135-concatenate-non-zero-digits-and-multiply-by-sum-i.java
class Solution {
    public long sumAndMultiply(int n) {
        int temp = n;
        ArrayList<Integer> dig = new ArrayList<>();
        while (temp > 0) {
            if (temp % 10 != 0)
                dig.add(temp % 10);
            temp /= 10;
        }
        Collections.reverse(dig);
        long sum = 0;
        for (int ele : dig)
            sum += ele;
        int x = 0;
        for (int ele : dig)
            x = x * 10 + ele;
        return x * 1L * sum; 
    }
}

File: 4135-concatenate-non-zero-digits-and-multiply-by-sum-i/4135-concatenate-non-zero-digits-and-multiply-by-sum-i.py
class Solution:
    def sumAndMultiply(self, n: int) -> int:
        def form(num: int) -> tuple:
            res = 0
            sum = 0
            num = list(str(num))
            for i in range(len(num)):
                if int(num[i]) != 0:
                    res = res * 10 + int(num[i])
                sum += int(num[i])
            
            return (res, sum)

        x, sum = form(n)
        return x * sum

        